packages/cadl-ranch-specs/http/authentication/apiKey/main.cadl

```diff
@@ -0,0 +1,28 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-azure-core";
+import "@azure-tools/cadl-ranch-expect";
+
+using Cadl.Http;
+using Azure.Core.Foundations;
+
+@serviceTitle("ApiKeyAuthentication")
```

Comment:
Just a tought not for your PR. Wonder if we could provide a new decorator in cadl-ranch-expect that does reduce a bunch of the service level boilerplate. Every spec needs pretty much the same 
- serviceVersion 
- server
- serviceTitle(take the namespace)

Reply:
Agree. Should be handy.

There is also a requirement that SDK emitter should fail on missing "doc" decorator (to enforce best-practice on Cadl file). https://github.com/Azure/autorest.csharp/issues/2621

Hence we might like any new Cadl spec here having full doc. I could ping dev that they would take best efforts to have their new/modified Cadl file having full doc.

Reply:
I did already say that on the chat but that requirement should absolutely not be there. The csharp generator should be generating a SDK if there is no doc, that's not the job of csharp generator to decide that. There is/will be linter enforcing doc for Azure services but if someone wants to generate CSharp without doc, they should be able to.

Reply:
Guess, need to talk with Laurent about this as seems like he signed up on this. It is ridiculous that if we are lacking doc the generator will just blow up, this would make generator pretty useless outside of Azure.

Reply:
Got it. I see.

It does feel like a lint/validation rule that should happen to Cadl file itself (with a tool maybe, but SDK emitter might not be the best one in this case).

Reply:
Yes exactly, right now with this behavior this mean someone will have a totally valid Cadl file with no error and then in the csharp(and seems like other generators) it will blow up giving no editor support for where the error is as well.

Reply:
Created an issue for the decorator https://github.com/Azure/cadl-ranch/issues/72